pkg/controller/statefulset/stateful_set_control.go

```diff
@@ -312,6 +315,34 @@ func (ssc *defaultStatefulSetControl) updateStatefulSet(
 			condemned = append(condemned, pods[i])
 		}
 		// If the ordinal could not be parsed (ord < 0), ignore the Pod.
+
+		for _, requested := range set.Spec.VolumeClaimTemplates {
+			pvcName := getPersistentVolumeClaimName(set, &requested, getOrdinal(pods[i]))
+			actual, err := ssc.client.CoreV1().PersistentVolumeClaims(set.Namespace).Get(pvcName, metav1.GetOptions{})
```

Comment:
You should retrieve the claims through the shared informers via a lister. See realStatefulPodControl